### PR TITLE
Add/Fix sausage requirements (casings, funnels)

### LINKED ISSUE #48953: Make raw sausage use stomach as casing and/or add synthetic casing and crafting
Wasteland sausage crafting requires stomach as a casing while regular sausage doesn't. It would be good to either make both use it/not use it and add some sort of synthetic casing with crafting to make it more realistic.

### PR DESCRIPTION
**Pull request by merrygin**
#### Summary
Bugfixes "All sausages need sausage casings and funnel now"

#### Purpose of change
Resolves #48953.
Some sausages needed neither casings nor stomach, some needed only stomach (wasteland sausage), and some used sausage casings item (bratwurst and hot dog). Also, sausage casings need a funnel, which doesn't make sense (should be in the sausage recipes isntead). Additionally, sausage casings require both stomachs AND plastic bags, which probably wasn't intended and doesn't make sense either. Realistically, it would be either / or.

#### Describe the solution
All sausages now need sausage casings, for more realism and consistency. 
Removed funnel requirement from the "sausage_casing" recipe. Instead, ALL sausage recipes now need a funnel, because that makes sense.
Changed the sausage casings recipe to not require stomachs AND plastic bags, but rather stomachs OR plastic bags to simulate natural vs. plastic casings (which probably was the intention before).

#### Describe alternatives you've considered
Considered additional tool requirements instead of or in addition to funnels, but decided against it for now for simplicitys sake. Also briefly considered different materials and different other aspects for casings recipe, but it seems fine like this for now, too.

#### Testing
Tested if recipes correct ingame.

#### Additional context
No additional context.

### CODE REVIEW

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -905,7 +906,7 @@
     "book_learn": [ [ "family_cookbook", 2 ], [ "scots_cookbook", 2 ] ],
     "batch_time_factors": [ 83, 3 ],
     "qualities": [ { "id": "CUT", "level": 1 }, { "id": "COOK", "level": 1 } ],
-    "tools": [ [ [ "surface_heat", 7, "LIST" ] ] ],
+    "tools": [ [ [ "surface_heat", 7, "LIST" ] ], [ ["funnel", -1] ]  ],
```
Comment (BrettDong) [inline data/json/recipes/recipe_food.json:909]:
```suggestion
    "tools": [ [ [ "surface_heat", 7, "LIST" ] ], [ [ "funnel", -1 ] ] ],
```

File: data/json/recipes/recipe_food.json [hunk 2]
```diff
@@ -6674,9 +6679,10 @@
     "autolearn": true,
     "batch_time_factors": [ 50, 3 ],
     "qualities": [ { "id": "CUT", "level": 1 }, { "id": "COOK", "level": 1 } ],
+    "tools": [ [ [ "funnel", -1] ] ],
```
Comment (BrettDong) [inline data/json/recipes/recipe_food.json:6682]:
```suggestion
    "tools": [ [ [ "funnel", -1 ] ] ],
```

**Review by BrettDong — COMMENTED** [review]
JSON formatting errors:

File: data/json/recipes/recipe_food.json (hunk 2 above)
Comment (Salty-Panda) [reply data/json/recipes/recipe_food.json:6682]: Shouldn't it allow to use makeshift funnel too?

File: data/json/recipes/recipe_food.json (hunk 1 above)
Comment (merrygin) [reply data/json/recipes/recipe_food.json:909]: Sorry again here, I'll proceed to make the changes as soon as possible.

**Comment by Maleclypse** [comment]
"Changed the sausage casings recipe to not require stomachs AND plastic bags, but rather stomachs OR plastic bags to simulate natural vs. plastic casings (which probably was the intention before)."  I'd rather you remove the plastic bags from making casings and make the sausage recipes have plastic bags as an alternative to using casings in the sausage making.  I made this recipe when I was new to contributions and I believe I'd added the plastic bag as a clean space which should actually be covered by the tools category.  https://butchersapprentice.wordpress.com/2010/11/26/hog-casings-from-scratch/ this was one of the places I used to create this recipe.

**Comment by merrygin** [comment]
> 
> 
> "Changed the sausage casings recipe to not require stomachs AND plastic bags, but rather stomachs OR plastic bags to simulate natural vs. plastic casings (which probably was the intention before)."
> I'd rather you remove the plastic bags from making casings and make the sausage recipes have plastic bags as an alternative to using casings in the sausage making. I made this recipe when I was new to contributions and I believe I'd added the plastic bag as a clean space which should actually be covered by the tools category. https://butchersapprentice.wordpress.com/2010/11/26/hog-casings-from-scratch/ this was one of the places I used to create this recipe.

Thanks for the feedback! So just to be clear: 
- throw out plastic bags from the casings recipe
- add plastic bags as choice to the sausage recipes as alternative to casings

Would you then also prefer a seperate recipe for artificial casings apart from the existing intestine casings? The point of the issue this PR is aiming at addressing was also to handle that distinction.

**Comment by Maleclypse** [comment]
> > "Changed the sausage casings recipe to not require stomachs AND plastic bags, but rather stomachs OR plastic bags to simulate natural vs. plastic casings (which probably was the intention before)."
> > I'd rather you remove the plastic bags from making casings and make the sausage recipes have plastic bags as an alternative to using casings in the sausage making. I made this recipe when I was new to contributions and I believe I'd added the plastic bag as a clean space which should actually be covered by the tools category. https://butchersapprentice.wordpress.com/2010/11/26/hog-casings-from-scratch/ this was one of the places I used to create this recipe.
> 
> Thanks for the feedback! So just to be clear:
> 
> * throw out plastic bags from the casings recipe
> * add plastic bags as choice to the sausage recipes as alternative to casings
> 
> Would you then also prefer a seperate recipe for artificial casings apart from the existing intestine casings? The point of the issue this PR is aiming at addressing was also to handle that distinction.

I'm sorry I'm just now seeing your questions.  Yes to both please.    I think if we make the first two changes we don't need a separate recipe.

**Review by Maleclypse — APPROVED** [review]
I've come to the conclusion the change I was asking for was semantic in nature.

**Comment by kwl01skz** [comment]
It is unreasonable to maintain the status quo.

Plastic bags cannot produce butchery refuse as byproducts.

It is suggested to delete butterfly refuse.

Even if the "stomach" is used, there is no need to produce butchery refuse. We have treated it when slaughtering.

**Pull request merged**